frontend/src/component/ui/Pagination.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { Component, ReactText } from 'react';
+
+import './Pagination.scss';
+
+interface IProps {
+  pageLimit: number;
+  totalRecords: number;
+  onPageChanged: (currentPage: number, pageLimit: number) => void;
+}
+
+interface IStates {
+  currentPage: number;
+}
+
+enum PageNavigator {
+  LEFT = 'LEFT',
+  RIGHT = 'RIGHT',
+  ELLIPSES = 'ELLIPSES'
+}
+
+const range = (from: number, to: number) => {
+  const result = [];
+  for (let i = from; i <= to; i++) {
+    result.push(i);
+  }
+  return result;
+};
```

Comment:
Let's move helpers to bottom so that the readers can focus the most important logic and scroll down when they want to read more about detailed implementation.